sync/src/block/extension.rs

```diff
@@ -138,34 +196,29 @@ impl Extension {
         }
     }
 
-    fn initial_state(client: Arc<Client>, snapshot_target: Option<(BlockHash, u64)>) -> State {
-        let (hash, num) = match snapshot_target {
-            Some(target) => target,
-            None => return State::Full,
-        };
-        let header = match client.block_header(&num.into()) {
-            Some(ref h) if h.hash() == hash => h.clone(),
-            _ => return State::SnapshotHeader(hash, num),
-        };
-        if client.block_body(&hash.into()).is_none() {
-            let parent_hash = header.parent_hash();
-            let parent =
-                client.block_header(&parent_hash.into()).expect("Parent header of the snapshot header must exist");
-            return State::SnapshotBody {
-                header,
-                prev_root: parent.transactions_root(),
+    fn move_state(&mut self) {
+        let next_state = self.state.next(&self.client);
+        cdebug!(SYNC, "Transitioning the state to {:?}", next_state);
+        if discriminant(&next_state) == discriminant(&State::Full) {
```

Comment:
You might want to
```rust
if let State::Full = next_state { ... }
```

Reply:
I copied this part from the existing code, and I think we should change that part too if we're fixing it.
Do you think your suggestion is better?

Reply:
I thought I've used it for matching complex enum. Not using `discriminant` is better for the short enum.

Reply:
It's used the same way as I used. ref: https://github.com/CodeChain-io/codechain/blob/snapshot/sync/src/block/extension.rs#L178
I'll change them altogether in a new commit then.